x-pack/legacy/plugins/reporting/server/lib/enqueue_job.ts

```diff
@@ -29,44 +29,43 @@ export type Job = EventEmitter & {
 export type EnqueueJobFn = <JobParamsType>(
   exportTypeId: string,
   jobParams: JobParamsType,
-  user: string,
-  headers: Record<string, string>,
-  request: RequestFacade
+  user: AuthenticatedUser | null,
+  context: RequestHandlerContext,
+  request: KibanaRequest
 ) => Promise<Job>;
 
 export function enqueueJobFactory(
   reporting: ReportingCore,
-  parentLogger: LevelLogger
+  setupDeps: ReportingInternalSetup
 ): EnqueueJobFn {
   const config = reporting.getConfig();
   const queueTimeout = config.get('queue', 'timeout');
   const browserType = config.get('capture', 'browser', 'type');
   const maxAttempts = config.get('capture', 'maxAttempts');
-
-  const logger = parentLogger.clone(['queue-job']);
+  const logger = setupDeps.logger.clone(['queue-job']);
```

Comment:
Since this clones the core logger vs one that gets passed in, log messages will not feature tags that were added by the caller. I think this was a good reason to leave `logger` as the last parameter in high-level functions like this, and have downstream code clone the "latest" logger.

Reply:
1cf1546